tools-ui/src/main/java/com/hedera/hashgraph/client/ui/SettingsPaneController.java

```diff
@@ -616,24 +633,38 @@ private void checkTransactionFee() {
 		defaultTransactionFee.setText(Utilities.setHBarFormat(transactionFee));
 	}
 
+	private Identifier calcNodeId(final String account) {
+		return Optional.ofNullable(networkChoicebox.getValue())
+				.filter(value -> value instanceof String)
+				.map(value -> (String) value)
+				.map(network -> Identifier.parse(account, network))
+				.orElseGet(() -> Identifier.parse(account));
+	}
+
 	private void checkNodeID() {
-		final var account = nodeIDTextField.getText();
-		final Identifier accountID;
+		final String account = nodeIDTextField.getText();
 		try {
-			accountID = Identifier.parse(account);
-			if (accountID.isValid()) {
-				controller.setDefaultNodeID(accountID.toReadableString());
-				nodeIDTextField.clear();
-				final Identifier defaultNodeID = Identifier.parse(controller.getDefaultNodeID());
-				final var s = defaultNodeID.toNicknameAndChecksum(controller.getAccountsList());
-				nodeIDTextField.setText(s);
-				settingScrollPane.requestFocus();
+			LOG.info("Node ID text field changed to: {}", account);
+			final Identifier calculatedIdentifier = calcNodeId(controller.getDefaultNodeID());
+			final String calculatedAccount = calculatedIdentifier.toNicknameAndChecksum(controller.getAccountsList());
+			if (Objects.equals(calculatedAccount, account)) {
+				controller.setDefaultNodeID(calculatedIdentifier.toReadableString());
 				accountIDErrorLabel.setVisible(false);
```

Comment:
This assumes that the current defaultNodeId is a valid account. Currently, I see no way to set the defaultNodeId if the account calculated for the nodId is not first validated, but perhaps in order to avoid potential issues in the future, it might not be best to assume this.

Reply:
yeah, if it is not valid an error is always shown. It was like that (with some problems) even before my change.